src/synapse/workload.py

```diff
@@ -870,28 +871,18 @@ def generate_nginx_config(container: ops.Container, main_unit_address: str) -> N
         container: Container of the charm.
         main_unit_address: Main unit address to be used in configuration.
     """
-    container.exec(
-        [
-            "cp",
-            "/etc/nginx/main_location.conf.template",
-            "/etc/nginx/main_location.conf",
-        ],
-    ).wait()
-    container.exec(
-        ["sed", "-i", f"s/main-unit/{main_unit_address}/g", "/etc/nginx/main_location.conf"],
-    ).wait()
-    container.exec(
-        [
-            "cp",
-            "/etc/nginx/abuse_report_location.conf.template",
-            "/etc/nginx/abuse_report_location.conf",
-        ],
-    ).wait()
-    container.exec(
-        [
-            "sed",
-            "-i",
-            f"s/main-unit/{main_unit_address}/g",
-            "/etc/nginx/abuse_report_location.conf",
-        ],
-    ).wait()
+    file_loader = FileSystemLoader(Path("./nginx_rock/etc"), followlinks=True)
+    env = Environment(loader=file_loader, autoescape=True)
+
+    # List of templates and their corresponding output files
+    templates = [
+        ("main_location.conf.j2", Path("main_location.conf")),
+        ("abuse_report_location.conf.j2", Path("abuse_report_location.conf")),
+    ]
+
+    for template_name, output_file in templates:
+        template = env.get_template(template_name)
+        output = template.render(main_unit_address=main_unit_address)
```

Comment:
Is possible that render fails?

Reply:
I'm not familiar with Jinja but I believe an exception could happen if main_unit_address is empty or an unexpected type, perhaps?
It seems an edge case but it would be interesting to check just in case.

Reply:
have done some investigating, and the method that generates the `main_unit_address`, `get_main_unit_address` will always return a `str` object. 

if this `str` is empty, then all cases of `main_unit_address` in the template file will be replaced by the empty string, which is expected behaviour, if not ideal. This should not happen though, because `get_main_unit_address()` builds a string using variables that should always be set.

I don't think that we need to catch any render exceptions, if an exception occurs it means something has gone horribly wrong elsewhere, and the error should have been caught somewhere more appropriate

Reply:
would appreciate your thoughts!

Reply:
Asked @nrobinaubertin  for input - I don't think that by catching any render errors will be useful for us - we cannot provide any more additional context than Jinja will already have. We are simply passing through a `str`, that even if `""` should still result in a successful render.